src/nupic/algorithms/Connections.cpp

```diff
@@ -55,6 +55,20 @@ void Connections::initialize(CellIdx numCells,
   maxSegmentsPerCell_ = maxSegmentsPerCell;
   maxSynapsesPerSegment_ = maxSynapsesPerSegment;
   iteration_ = 0;
+  nextEventToken_ = 0;
+}
+
+UInt32 Connections::subscribe(ConnectionsEventHandler* handler)
+{
+  UInt32 token = nextEventToken_++;
+  eventHandlers_[token] = handler;
+  return token;
+}
+
+void Connections::unsubscribe(UInt32 token)
```

Comment:
Fair point. But then:

- This class would have to store copies of strings.
- The caller would have to be careful not to use strings that are used elsewhere. When I grab HTM traces, I do it via a set of independent classes that are unaware of each other. It'd be easy for one of these classes to accidentally stomp on another class's event handler. To avoid this, each tracing class would have to generate a string and store it.

This "token" approach mimics javascript's `setTimeout` and `setInterval` API.